test/src/unittests/tonal/test_pitchsaliencefunction.py

```diff
@@ -0,0 +1,378 @@
+
+    def testSinglePeak(self):        
+        # Provide a single input peak with a unit magnitude at the reference frequency, and 
+        # validate that the output salience function has only one non-zero element at the first bin.       
+        # N.B: default value for bin Resolution is 10.
+        freq_speaks = [55] 
+        mag_speaks = [1] 
+        outputLength  = 600 # calculated by fiveOctaveFullRange/binResolution = 6000/10        
+         # Length of the non-zero values for this Pitch Salience = 11
+        expectedPitchSalience = [1.0000000e+00, 9.7552824e-01, 9.0450847e-01, 7.9389262e-01, 6.5450847e-01,
+        5.0000000e-01, 3.4549147e-01, 2.0610739e-01, 9.5491491e-02, 2.4471754e-02, 3.7493994e-33]
+
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        self.assertEqual(len(calculatedPitchSalience), outputLength)       
+        # Check the first 11 elements. The first element has value "1".
+        # The next returned 10 non-zero values decreasing in magnitude, should match "expected" above.
+        self.assertAlmostEqualVector(calculatedPitchSalience[:11], expectedPitchSalience, 8)
+        # Check remaining elements are zeros.
+        self.assertEqualVector(calculatedPitchSalience[12:outputLength], zeros(outputLength-(12)))
+
+    def testSinglePeakNonDefaultBR(self):   
+        # Same as above, but tweaking the Bin Resolution to ensure output length is consistant
+        # Larger bin resolution reduces the number of non zero values in salience function
+        binResolution = 40
+        freq_speaks = [55] 
+        mag_speaks = [1] 
+        outputLength  = int (6000/binResolution)
+        # Length of the non-zero values for this Pitch Salience = 3        
+        expectedPitchSalience = [1.0000000e+00, 5.0000000e-01, 3.7493994e-33]
+
+        calculatedPitchSalience = PitchSalienceFunction(binResolution=binResolution)(freq_speaks,mag_speaks)
+        self.assertEqual(len(calculatedPitchSalience), outputLength)
+        # Check the first 3 elements. The first has value "1".
+        # The next returned 3 non-zero values decreasing in magnitude, should match "expected" above.        
+        self.assertAlmostEqualVector(calculatedPitchSalience[:3], expectedPitchSalience, 8)
+        # Check remaining elements are zeros
+        self.assertEqualVector(calculatedPitchSalience[4:outputLength], zeros(outputLength-4))
+
+    def testSinglePeakLowCompression(self):        
+        # Provide a single input peak with a unit magnitude at the reference frequency, and 
+        # validate that the output salience function has only one non-zero element at the first bin.             
+        freq_speaks = [55] 
+        mag_speaks = [1] 
+        outputLength  = 600        
+         # Length of the non-zero values for this Pitch Salience = 11
+        expectedPitchSalience = [1.0000000e+00, 9.7552824e-01, 9.0450847e-01, 7.9389262e-01, 6.5450847e-01,
```

Comment:
This test should be updated. Magnitude compression is `pow(magnitude, magnitudeCompression)` so the output is the same because the input magnitude is 1. Test using a smaller input magnitude value. The `expectedPitchSalience` will change.